src/inc/quic_platform_posix.h

```diff
@@ -517,21 +548,21 @@ CxPlatPoolInitialize(
     CxPlatLockInitialize(&Pool->Lock);
     Pool->ListDepth = 0;
     CxPlatZeroMemory(&Pool->ListHead, sizeof(Pool->ListHead));
+    Pool->Allocate = Allocate;
+    Pool->Free = Free;
     UNREFERENCED_PARAMETER(IsPaged);
 }
 
 inline
 void
-CxPlatPoolUninitialize(
+CxPlatPoolInitialize(
+    _In_ BOOLEAN IsPaged,
+    _In_ uint32_t Size,
+    _In_ uint32_t Tag,
     _Inout_ CXPLAT_POOL* Pool
     )
 {
-    CXPLAT_POOL_HEADER* Entry;
-    while ((Entry = (CXPLAT_POOL_HEADER*)CxPlatListPopEntry(&Pool->ListHead)) != NULL) {
-        CXPLAT_DBG_ASSERT(Entry->SpecialFlag == CXPLAT_POOL_FREE_FLAG);
-        CxPlatFree(Entry, Pool->Tag);
-    }
-    CxPlatLockUninitialize(&Pool->Lock);
+    CxPlatPoolInitializeEx(IsPaged, Size, Tag, NULL, NULL, Pool);
```

Comment:
Instead of NULLs and always doing if checks in every alloc/free, should we have defaults that call what we want?

Reply:
I might back out this change if I don't get as far as zerocopy TX, which is the only use case for this model.

Reply:
I reverted this for now, but I'm actually not sure whether a default handler (through an indirect function pointer) is higher performance than a NULL check and inline fallback. The answer is probably "it depends".